src/org/opendatakit/briefcase/export/SubmissionMetaData.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2018 Nafundi
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.opendatakit.briefcase.export;
+
+import static java.util.stream.Collectors.toList;
+
+import java.time.OffsetDateTime;
+import java.util.List;
+import java.util.Optional;
+import org.opendatakit.briefcase.model.ParsingException;
+import org.opendatakit.briefcase.reused.Optionals;
+
+/**
+ * This class holds a form's metadata. Instances of this class are
+ * generated while parsing submissions and they are used to read
+ * each value only once.
+ * <p>
+ * All its members are lazily evaluated to avoid unnecessary parsing
+ */
+class SubmissionMetaData {
+  private final XmlElement root;
+  // All these members are not final because they're lazily evaluated
+  private String formId;
+  private Optional<String> instanceId;
+  private Optional<String> version;
+  private Optional<OffsetDateTime> submissionDate;
+  private Optional<String> encryptedXmlFile;
+  private Optional<String> base64EncryptedKey;
+  private Optional<String> encryptedSignature;
+  private List<String> mediaNames;
+
+  /**
+   * Main constructor for {@link SubmissionMetaData} class. It takes
+   * an {@link XmlElement} to act as the root element which will be
+   * queries for the different values this class can offer.
+   */
+  SubmissionMetaData(XmlElement root) {
+    this.root = root;
+  }
+
+  /**
+   * Returns the submission date, located at the root node's "submissionDate" attribute.
+   * <p>
+   * The value gets mapped to an {@link OffsetDateTime}
+   */
+  Optional<OffsetDateTime> getSubmissionDate() {
+    if (submissionDate == null)
+      submissionDate = root.getAttributeValue("submissionDate")
+          .map(iso8601DateTime -> {
+            // Sometimes we can get just an ISO8601-ish string lacking of
+            // the minutes part of the offset block. Fix that.
+            return iso8601DateTime.length() - iso8601DateTime.lastIndexOf("+") == 3
+                ? iso8601DateTime + ":00"
+                : iso8601DateTime;
+          })
+          .map(OffsetDateTime::parse);
```

Comment:
Do you need to catch DateTimeParseException from parse? How confident are you the date strings will be in the right format?